File: PAT/BLevel/1002/1002.go
// 1002 写出这个数
// auguralpha
// 2020-03-24

package main

import "fmt"

func main() {
	var n string
	var cnt int = 0
	var str = [10]string{"ling", "yi", "er", "san", "si", "wu", "liu", "qi", "ba", "jiu"}
	var ans [10]int
	_, _ = fmt.Scanf("%s", &n)
	for i := 0; i < len(n); i ++ {
		cnt += int(n[i] - '0')
	}
	var m int = 0
	for cnt != 0 {
		ans[m] = cnt % 10
		m ++
		cnt /= 10
	}

	for i := m - 1; i >= 0; i -- {
		if i != 0 {
			fmt.Printf("%s ", str[ans[i]])
		} else {
			fmt.Printf("%s\n", str[ans[i]])
		}
	}
}

File: PAT/BLevel/1002/1002.cpp
// 1002 写出这个数
// auguralpha
// 2019-02-11
#include <cstdio>
#include <string>
#include <iostream>
using namespace std;
const int maxn = 10;
string str[10] = {"ling", "yi", "er", "san", "si", "wu", "liu", "qi", "ba", "jiu"};
int ans[maxn];

int main() {
    string n;
    int cnt = 0;
    cin >> n;
    for(int i = 0; i < n.size(); ++i) {
        cnt += n[i] - '0';
    }
    int m = 0;
    while(cnt != 0) {
        ans[m++] = cnt % 10;
        cnt /= 10;
    }
    for(int i = m-1; i >= 0; --i) {
        if(i != 0) {
            cout << str[ans[i]] << " ";
        }
        else {
            cout << str[ans[i]] << endl;
        }
    }
    return 0;
}
